evm/src/cross_table_lookup.rs

```diff
@@ -423,19 +423,22 @@ impl<F: Field> Filter<F> {
     }
 }
 
-/// A `Table` with a linear combination of columns and a filter.
-/// `filter` is used to determine the rows to select in `Table`.
-/// `columns` represents linear combinations of the columns of `Table`.
+/// An alias for `usize`, to represent the index of a STARK table in a multi-STARK setting.
+pub(crate) type TableIdx = usize;
+
+/// A `table` index with a linear combination of columns and a filter.
+/// `filter` is used to determine the rows to select in `table`.
+/// `columns` represents linear combinations of the columns of `table`.
 #[derive(Clone, Debug)]
 pub(crate) struct TableWithColumns<F: Field> {
-    table: Table,
+    table: TableIdx,
```

Comment:
What about `:T where T: Into` , or another ad-hoc trait. That should work for any enum for representing the tables. Though you need to add a type parameter

Reply:
This would require you to add everywhere calls to `table.into()` which I initially didn't want, or you'd need additional traits (from the top of my head, you'd need `Into<usize> + PartialEq + Eq + Debug` but even with this you'd still need to call conversions to `usize` here and there, so overall I'm not sure this brings much (and the user still needs to implement `From<Foo> for usize`, similarly to the current requirement to implement `Deref`.

Reply:
Nevermind, I was assuming enums implement Into<usize>. But still, you may want to use whatever you want for your tables (an enum for example) and here you just want to be able to derive an index for each table

Reply:
This would require you to add everywhere calls to `table.into()` which I initially didn't want, or you'd need additional traits (from the top of my head, you'd need `Into<usize> + PartialEq + Eq + Debug` but even with this you'd still need to call conversions to `usize` here and there, so overall I'm not sure this brings much (and the user still needs to implement `From<Foo> for usize`, similarly to the current requirement to implement `Deref`.

Reply:
> But still, you may want to use whatever you want for your tables (an enum for example) and here you just want to be able to derive an index for each table

This is still possible with the current approach though (I kept the `Table` enum), you just need to implement `Deref`.

Reply:
Ok, I see. then it's ok I think. I was just confused with the "hacky" implementation of deref .